src/redux/selectors/index.js

```diff
@@ -203,6 +203,41 @@ const filteredItemsInDateRangeSelector = createSelector(
   },
 );
 
+export const selectedRecordsGroupedByTypeSelector = createSelector(
+  [filteredItemsInDateRangeSelector, activeCollectionResourceTypeSelector],
+  (items, selectedResourceType) => {
+    const typeMap = items
+      .reduce((acc, item) => {
+        const { type, subType } = item;
+        return produce(acc, (draft) => {
+          // eslint-disable-next-line no-param-reassign
+          draft[type] = draft[type] ?? {};
+          // eslint-disable-next-line no-param-reassign
+          draft[type][subType] = draft[type][subType] ?? [];
+          draft[type][subType].push(item.id);
+        });
+      }, {});
+
+    const sortedGroupedResources = Object.entries(typeMap).reduce((acc, [type, subTypes]) => {
+      const subTypesArray = Object.entries(subTypes)
+        .map(([subType, recordIds]) => ({ subType, recordIds }))
+        .sort(({ subType: s1 }, { subType: s2 }) => (
+          (s1.toLowerCase() < s2.toLowerCase()) ? -1 : 1));
```

Comment:
this might actually be more clear if `typeObject.subTypes` is directly assigned to `typeObject.subTypes`